#! /bin/sh -e 
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Description: Disables legacy serial driver on powermacs.
## DP: Patch author: Sven Luther <luther@debian.org> 
## DP: Patch author: adapted from the SuSE kernel tree.
## DP: Upstream status: workaround hack waiting for a clean legacy device solution.

diff -aurN a/drivers/serial/8250.c b/drivers/serial/8250.c
--- a/drivers/serial/8250.c	2005-06-17 15:48:29.000000000 -0400
+++ b/drivers/serial/8250.c	2005-06-18 12:05:39.000000000 -0400
@@ -46,6 +46,10 @@
 
 #include "8250.h"
 
+#ifdef CONFIG_PPC_MULTIPLATFORM
+#include <asm/processor.h>
+#endif
+
 /*
  * Configuration:
  *   share_irqs - whether we pass SA_SHIRQ to request_irq().  This option
@@ -2188,6 +2192,12 @@
 
 static int __init serial8250_console_init(void)
 {
+#ifdef CONFIG_PPC_MULTIPLATFORM
+	if(_machine == _MACH_Pmac) {
+		printk("%s: nothing to do on PowerMac\n",__FUNCTION__);
+		return -ENODEV;
+	}
+#endif
 	serial8250_isa_init_ports();
 	register_console(&serial8250_console);
 	return 0;
@@ -2491,6 +2501,12 @@
 {
 	int ret, i;
 
+#ifdef CONFIG_PPC_MULTIPLATFORM
+	if(_machine == _MACH_Pmac) {
+		printk("%s: nothing to do on PowerMac\n",__FUNCTION__);
+		return -ENODEV;
+	}
+#endif
 	printk(KERN_INFO "Serial: 8250/16550 driver $Revision: 1.90 $ "
 		"%d ports, IRQ sharing %sabled\n", (int) UART_NR,
 		share_irqs ? "en" : "dis");
